src/components/TodoCreationForm.js

```diff
@@ -1,27 +1,52 @@
 import React from "react";
+import DatePicker from "react-datepicker";
+import Button from "./Library/Button";
 
 export default class TodoCreationForm extends React.Component {
   state = {
-    taskName: ""
+    todoName: "",
+    dueDate: new Date()
   }
 
+  handleDueDateChange = date => {
+    this.setState({
+      dueDate: date
+    });
+  };
+
   render() {
     return (
       <form
+        className="todo-creation-form"
         onSubmit={e => {
           e.preventDefault();
-          if (this.state.taskName === "") return;
-          this.props.onCreate(this.state.taskName)
-          this.setState({taskName: ""})
+          if (this.state.todoName === "") return;
+          this.setState({todoName: ""})
         }}
       >
         <input
           className="input is-primary is-small"
           autoFocus
-          placeholder="Add a task (hit Enter to validate)"
-          value={this.state.taskName}
-          onChange={e => this.setState({taskName: e.target.value})}
+          placeholder="Add a todo (hit Enter to validate)"
+          value={this.state.todoName}
+          onChange={e => this.setState({todoName: e.target.value})}
+        />
+        <DatePicker
+          selected={this.state.dueDate}
+          onChange={this.handleDueDateChange}
+          className="input date-picker-input is-primary is-small"
         />
+        <Button
+          type="button"
+          color="success"
+          size="small"
+          disabled={this.state.todoName === ""}
+          onClick={() => this.props.onCreate(this.state.todoName, this.state.dueDate)}
+        >
+          <span className="icon">
+            <i className="fas fa-check-circle"></i>
+          </span>
+        </Button>
```

Comment:
Nice, you reproduced the button in my screen :). 
Some comments though:
- The button should be of type "submit" to indicate that this button should trigger the form `onSubmit` when clicked.
- Move the `onClick` code to the form `onSubmit`. You should not have different codes for the submit action. Only one. I like to put everything in the `onSubmit` and simply add the type `"submit"`
 on the button.
- The button should be disabled when the todoName is empty or the dueDate is empty. The form should not call `this.props.onCreate` when both fields are not completed. I suggest that you create a const `disabled` at the start of the render method and use it.  The disabled const would be defined like so `const disabled = !this.state.todoName || !this.state.dueDate`. You can also do the reversed thing if you prefer : `const canSubmit = !!this.state.todoName || !!this.state.dueDate` (the `!!` converts the value to a boolean, either true or false depending on the value)
For info, a string will evaluate to false when empty. A null value will also evaluate to false.

Reply:
For the last part, I had already done that: "disabled={this.state.todoName === "" || this.state.dueDate === null}" -> is it correct?